cmd/minishift/cmd/start.go

```diff
@@ -315,58 +329,6 @@ func automountHostfolders(driver drivers.Driver) {
 	}
 }
 
-// Set Docker Proxy
-func setDockerProxy() {
-	if viper.IsSet(httpProxy) {
-		dockerEnv = append(dockerEnv, fmt.Sprintf("HTTP_PROXY=%s", viper.GetString(httpProxy)))
-	}
-	if viper.IsSet(httpsProxy) {
-		dockerEnv = append(dockerEnv, fmt.Sprintf("HTTPS_PROXY=%s", viper.GetString(httpsProxy)))
-	}
-	if viper.IsSet(noProxyList) {
-		dockerEnv = append(dockerEnv, fmt.Sprintf("NO_PROXY=%s,%s", updateNoProxyForDocker(),
-			viper.GetString(noProxyList)))
-	} else if viper.IsSet(httpProxy) || viper.IsSet(httpsProxy) {
```

Comment:
@praveenkumar here is a question for you. I basically tried to keep all of the semantics of the existing code, except for the part here. In your code, if --no-proxy is specified only this value is used for NO_PROXY and the default values are not applied. I think this is actually wrong and a bug which just had not surfaced yet. IMO, the default no proxy list needs to be added as soon as a proxy is used and they are additional to any explicitly specified no proxy hosts. WDYT?